grr/server/grr_response_server/gui/ui/components/client_details/client_details.ng.html

```diff
@@ -22,274 +22,233 @@
+    <table class="section">
+      <tr *ngIf="client.osInfo.machine">
+        <td class="description">CPU Architecture</td>
+        <td class="value">{{client.osInfo.machine}}</td>
+      </tr>
+      <tr *ngIf="client.memorySize">
+        <td class="description">Memory size</td>
+        <td class="value"><human-readable-size [size]="client.memorySize"></human-readable-size></td>
+      </tr>
+    </table>
+
+    <br>
+    <div class="section-title mat-h3">Network Interfaces</div>
+    <table class="section">
+      <ng-template ngFor let-interface [ngForOf]="client.networkInterfaces.slice(0, currentNumInterfacesShown)" let-isLast="last">
         <tr>
-          <td class="description">Client ID</td>
-          <td class="value">{{client.clientId}}</td>
-        </tr>
-        <tr *ngIf="client.knowledgeBase.fqdn">
-          <td class="description">FQDN</td>
-          <td class="value">{{client.knowledgeBase.fqdn}}</td>
-        </tr>
-        <tr *ngIf="(client$ | async)?.lastSeenAt as lastSeenAt">
-          <td class="description">Last seen</td>
-          <td class="value"><timestamp [date]="lastSeenAt"></timestamp></td>
-        </tr>
-        <tr *ngIf="(client$ | async)?.lastClock as lastClock">
-          <td class="description">Last clock</td>
-          <td class="value"><timestamp [date]="lastClock"></timestamp></td>
+          <td class="description">Interface name</td>
+          <td class="value">{{interface.interfaceName}}</td>
         </tr>
-        <tr *ngIf="(client$ | async)?.lastBootedAt as lastBoot">
-          <td class="description">Last boot</td>
-          <td class="value"><timestamp [date]="lastBoot"></timestamp></td>
+        <tr>
+          <td class="description">MAC</td>
+          <td class="value">{{interface.macAddress}}</td>
         </tr>
         <tr>
-          <td class="description">Labels</td>
+          <td class="description">IP addresses</td>
           <td class="value">
-            <mat-chip-list>
-              <mat-chip *ngFor="let label of (client$ | async)?.labels" class="chip">
-                {{label.name}}
-              </mat-chip>
-            </mat-chip-list>
+            <div *ngFor="let address of interface.addresses">
+              {{address.ipAddress}}
+            </div>
           </td>
         </tr>
-      </table>
+        <br *ngIf="!isLast">
+      </ng-template>
+      <div class="empty-section" *ngIf="client.networkInterfaces.length === 0">None.</div>
+    </table>
+    <button
+      *ngIf="getAccordionButtonState(client.networkInterfaces.length, currentNumInterfacesShown, INITIAL_NUM_INTERFACES_SHOWN) === 'show-more'"
+      mat-button color="primary" (click)="currentNumInterfacesShown = client.networkInterfaces.length">
+        Show {{client.networkInterfaces.length - currentNumInterfacesShown}} more network interface<ng-template [ngIf]="client.networkInterfaces.length - currentNumInterfacesShown > 1">s</ng-template><mat-icon>keyboard_arrow_down</mat-icon>
```

Comment:
Wow, we should probably invest in some library for word pluralization.

Reply:
I just searched about this, angular actually has a directive for this, called [NgPlural](https://angular.io/api/common/NgPlural). Other solution would be to create a custom pipe that would change the word depending on the number of elements. IMO, the first method is cleaner, and allows for better customization in general.